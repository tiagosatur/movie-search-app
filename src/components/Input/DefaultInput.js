import React from 'react';
import styled, { css } from 'styled-components';
import { colorPallete, rem } from '../../style';

const DefaultInput = props => {
  const {
    classes,
    placeholder,
    id,
    name,
    value,
    disabled,
    type,
    handleKeyUp,
    handleChange,
    submit,
  } = props;

  return (
    <StyledDefaultInput
      type={type}
      id={id}
      name={name}
      value={value}
      className={`input ${classes ? classes : ''}`}
      placeholder={placeholder}
      disabled={disabled}
      onKeyUp={handleKeyUp}
      onChange={handleChange}
      onSubmit={submit}
    />

  );
};

const inputBaseStyle = css`
    align-items: center;
    border: ${rem(1)} solid ${colorPallete.light['80']};
    border-radius: ${rem(2)};
    box-shadow: none;
    display: inline-flex;
    font-size: ${rem(16)};
    height: ${rem(32)};
    justify-content: flex-start;
    line-height: $control-line-height;
    padding: ${rem(8)};
`;

const StyledDefaultInput = styled.input`
    ${inputBaseStyle}

    border-left-color: ${({ errors, name }) => {
    return errors && errors[name]
      ? 'red'
      : colorPallete.light['80'];
  }};
  border-left-width: ${({ errors, name }) =>
    errors && errors[name] ? rem(5) : rem(1)};

  &:hover,
  &:focus {
    border-color: ${({ errors, name }) => {
      return errors && errors[name]
        ? 'red'
        : colorPallete.light['87'];
    }};
    border-left-width: ${({ errors, name }) =>
      errors && errors[name] ? rem(5) : rem(1)};
  }
`;

export default DefaultInput;
