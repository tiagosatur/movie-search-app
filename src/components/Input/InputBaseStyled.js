import { css } from 'styled-components';
import { colorPallete, rem } from '../../style';

export const InputBaseStyled = css`
    align-items: center;
    border: ${rem(1)} solid ${colorPallete.light['80']};
    border-radius: ${rem(2)};
    box-shadow: none;
    display: inline-flex;
    font-size: ${rem(16)};
    height: ${rem(32)};
    justify-content: flex-start;
    line-height: $control-line-height;
    padding: ${rem(8)};
    transition: all .4s;
    width: 100%;
`;
