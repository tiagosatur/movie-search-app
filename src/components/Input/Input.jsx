import React from 'react';
import styled from 'styled-components';
import { colorPallete } from '../../style';
import rem from '../../style/mixins/rem';
import { InputBaseStyled } from './InputBaseStyled'

const Input = props => {
  const {
    classes,
    placeholder,
    id,
    name,
    disabled,
    type,
    register,
    errors,
    handleKeyUp,
  } = props;
  

  return (
    <Field>
      <StyledInput
        type={type ? type : 'text'}
        id={id && id}
        name={name}
        ref={register}
        errors={errors}
        className={`input ${classes ? classes : ''}`}
        placeholder={placeholder}
        disabled={disabled}
        onKeyUp={handleKeyUp}
      />
      {errors[name] ? (
        <StyledSpan errors={errors} name={name} className='field-error'>
          {errors[name] && errors[name].message}
        </StyledSpan>
      ) : null}
    </Field>
  );
};

const Field = styled.div`
    margin-bottom: ${rem(16)};
`;

const StyledInput = styled.input`
  ${ InputBaseStyled }

  border-left-color: ${({ errors, name }) => {
    return errors && errors[name]
      ? 'red'
      : colorPallete.light['87'];
  }};
  border-left-width: ${({ errors, name }) =>
    errors && errors[name] ? rem(5) : rem(1)};

  &:hover,
  &:focus {
    border-color: ${({ errors, name }) => {
      return errors && errors[name]
        ? 'red'
        : colorPallete.light['87'];
    }};
    border-left-width: ${({ errors, name }) =>
      errors && errors[name] ? rem(5) : rem(1)};
  }
`;

const StyledSpan = styled.span`
  color: red;
  display: ${({ errors, name }) => {
    return errors[name] && errors[name].message ? 'flex' : 'none';
  }};
  font-size: ${rem(14)};
  padding: ${rem(4)} ${rem(8)} 0 ${rem(8)};
  transition: all 0.5s;
`;

export default Input;
