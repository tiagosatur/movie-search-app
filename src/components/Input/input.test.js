import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './Input';
import { stub } from 'sinon';

const register = stub();

describe('Input tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Input 
        name='username' 
        id='input-username'
        register={
          register({ required: true })
        }
        errors={{ username: { message: 'x', pattern: 'x' }}} />
    ).dive();
  });
  
  it('should render the Input element', () => {
    expect(wrapper.find('.input').exists()).toBe(true);
  });

  it('should call Input snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})