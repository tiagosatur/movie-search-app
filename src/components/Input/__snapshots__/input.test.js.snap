// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests should call Input snapshot correctly 1`] = `
<div
  className="sc-bdVaJa gBePXc"
>
  <input
    className="input  sc-bwzfXH bqcSTh"
    id="input-username"
    name="username"
    type="text"
  />
  <span
    className="field-error sc-htpNat iRpXGe"
    name="username"
  >
    x
  </span>
</div>
`;
